server/src/test/java/org/elasticsearch/common/settings/ScopedSettingsTests.java

```diff
@@ -1221,4 +1256,54 @@ public String getKey(final String key) {
                 equalTo(oldSetting.get(settings).stream().map(s -> "new." + s).collect(Collectors.toList())));
     }
 
+    public void testUpdateOfValidationDependantSettings() {
+        ClusterSettings service = new ClusterSettings(Settings.EMPTY, new HashSet<>(Arrays.asList(SETTING_FOO_LOW, SETTING_FOO_HIGH)));
+
+        Settings.Builder target = Settings.builder();
+        Settings.Builder updates;
+        boolean updated;
+
+        updates = Settings.builder();
+        updated = service.updateSettings(
+            Settings.builder().put(SETTING_FOO_LOW.getKey(), 20).build(),
+            target,
+            updates,
+            "transient"
+        );
+        assertTrue(updated);
+        assertThat(target.get(SETTING_FOO_LOW.getKey()), equalTo("20"));
+        assertThat(updates.get(SETTING_FOO_LOW.getKey()), equalTo("20"));
+
+        updates = Settings.builder();
+        updated = service.updateSettings(
+            Settings.builder().put(SETTING_FOO_LOW.getKey(), 20).build(),
+            target,
+            updates,
+            "transient"
+        );
+        assertFalse(updated);
+        assertThat(target.get(SETTING_FOO_LOW.getKey()), equalTo("20"));
+        assertThat(updates.keys(), empty());
+
+        updates = Settings.builder();
+        updated = service.updateSettings(
+            Settings.builder().put(SETTING_FOO_HIGH.getKey(), 40).build(),
+            target,
+            updates,
+            "transient"
+        );
+        assertTrue(updated);
+        assertThat(target.get(SETTING_FOO_LOW.getKey()), equalTo("20"));
+        assertThat(target.get(SETTING_FOO_HIGH.getKey()), equalTo("40"));
+        assertNull(updates.get(SETTING_FOO_LOW.getKey()));
+        assertThat(updates.get(SETTING_FOO_HIGH.getKey()), equalTo("40"));
+
+        Exception exception = expectThrows(IllegalArgumentException.class, () -> service.updateSettings(
+            Settings.builder().put(SETTING_FOO_HIGH.getKey(), 10).build(),
+            target,
+            Settings.builder(),
+            "transient"
+        ));
+        assertThat(exception.getMessage(), equalTo("[high]=10 is lower than [low]=20"));
```

Comment:
Could we also have a case that shows that if `low` is set below `10` then `high` can also be set below `10`, to verify that we're not validating against both the default and the actual values?